src/osp/OSPApplication.h

```diff
@@ -39,12 +40,32 @@ class OSPApplication
 
     OSPApplication();
 
-    std::vector<Package>& debug_get_packges() { return m_packages; };
+    //std::vector<Package>& debug_get_packges() { return m_packages; };
+
+    /**
+     * Add a resource package to the application
+     *
+     * The package should be populated externally, then passed via rvalue
+     * reference so the contents can be moved into the application resources
+     *
+     * @param p [in] The package to add
+     */
+    void debug_add_package(Package&& p);
+
+    /**
+     * Get a resource package by prefix name
+     *
+     * @param prefix [in] The short prefix name of the package
+     * @return The resource package
+     */
+    Package& debug_get_package(resprefix_t prefix);
+
+    size_t debug_num_packages() const { return m_packages.size(); }
 
     universe::Universe& get_universe() { return m_universe; }
 
 private:
-    std::vector<Package> m_packages;
+    std::map<resprefix_t, Package> m_packages;
```

Comment:
std::map<resprefix_t, Package, std::less<>>

Reply:
Is this not the default behavior?

Reply:
This "transparent comparator" allows string views to access the map. If I understand correctly, it will allow any type comparable to an std::string.